### PR TITLE
Add LoadingOverlay for Component And Bundle Loading

### PR DESCRIPTION
**Pull request by oshi97**
This PR adds the LoadingOverlay components, which will provide a loading screen while user components and
the rest of the application are being loaded over the network.

In the future we can add additional information to this screen regarding hitting the management api and loading
in localData files, if necessary.

J=SLAP-2895
TEST=manual

### CODE REVIEW

File: packages/studio-ui/src/utils/getFunctionComponent.ts [hunk 1]
```diff
@@ -1,15 +1,7 @@
-import { FunctionComponent } from "react";
 import { ImportType } from "../store/models/ImportType";
 
 export default function getFunctionComponent(
-  importedValue: Record<string, unknown>,
-  name: string
+  importedValue: Record<string, unknown>
 ): ImportType | undefined {
-  if (typeof importedValue[name] === "function") {
```
Comment (oshi97) [inline packages/studio-ui/src/utils/getFunctionComponent.ts:8]: We don't support components that are not default exported, I think we probably used to which is why we had this logic but haven't checked. Possibly is related to the external components support that Tristan did that we had to cut

File: packages/studio-ui/src/AppWithLazyLoading.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          {renderComponentLoadingProgress(
+            loadedCount,
+            totalCount,
+            componentsLoaded
+          )}
+          {renderBundleMessage(appLoaded)}
+        </>
+      }
+    >
+      <Suspense>
```
Comment (alextaing) [inline packages/studio-ui/src/AppWithLazyLoading.tsx:38]: if we have the loading overlay, do we still need the suspense?  Or does it do something else other than display a fallback UI?

Comment (alextaing) [reply packages/studio-ui/src/AppWithLazyLoading.tsx:38]: is it just so everything in the app renders at once?

File: packages/studio-ui/.size-limit.cjs [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module.exports = [
   {
-    path: "lib/src/index.js",
+    path: "lib/**/*.js",
```
Comment (alextaing) [inline packages/studio-ui/.size-limit.cjs:3]: more of a question for me, but what is the reason behind this change?

File: packages/studio-ui/src/utils/loadComponents.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * Load all of the user's components into the store.
+ */
+export default function loadComponents(): Promise<void>[] {
+  const { UUIDToFileMetadata, setImportedComponent, UUIDToImportedComponent } =
+    useStudioStore.getState().fileMetadatas;
+
+  const componentImportPromises = Object.values(UUIDToFileMetadata).map(
+    async (fileMetadata) => {
+      if (UUIDToImportedComponent.hasOwnProperty(fileMetadata.metadataUUID)) {
```
Comment (alextaing) [inline packages/studio-ui/src/utils/loadComponents.tsx:14]: also for my edification, would something like `if (fileMetadata.metadataUUID in UUIDToImportedComponent)` be more readable, or is there a functional difference for this case?

Comment (alextaing) [reply packages/studio-ui/src/utils/loadComponents.tsx:14]: its OK, found an article: https://masteringjs.io/tutorials/fundamentals/hasownproperty

File: packages/studio-ui/.size-limit.cjs (hunk 3 above)
Comment (oshi97) [reply packages/studio-ui/.size-limit.cjs:3]: Adding a dynamic import to the bundle adds additional js files to studio-ui/lib that don't live under the src folder

File: packages/studio-ui/src/utils/loadComponents.tsx (hunk 4 above)
Comment (oshi97) [reply packages/studio-ui/src/utils/loadComponents.tsx:14]: oh yeah distinction is small and 99% of the time unimportant

File: e2e-tests/__screenshots__/darwin/add-element.spec.ts/can-add-a-Footer-component-1.png
Comment (alextaing) [inline e2e-tests/__screenshots__/darwin/add-element.spec.ts/can-add-a-Footer-component-1.png:1]: Not related to the changes in this PR, but I think at some point we lost our local data or something b/c the hours component looks like it broke.  Should we look into adding this back at some point?  It also looks like our styling got messed up too for the BusinessInfo container.

Comment (alextaing) [reply e2e-tests/__screenshots__/darwin/add-element.spec.ts/can-add-a-Footer-component-1.png:1]: other than that, lgtm!

**Review by alextaing — DISMISSED** [review]

File: e2e-tests/__screenshots__/darwin/add-element.spec.ts/can-add-a-Footer-component-1.png
Comment (oshi97) [reply e2e-tests/__screenshots__/darwin/add-element.spec.ts/can-add-a-Footer-component-1.png:1]: I regenerated the local data!

Comment (alextaing) [reply e2e-tests/__screenshots__/darwin/add-element.spec.ts/can-add-a-Footer-component-1.png:1]: should we regen screenshots, or do you want to just do that as a part of the larger screenshot item?

File: packages/studio-ui/src/AppWithLazyLoading.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      </div>
+    </>
+  );
+}
+
+function renderBundleMessage(appLoaded: boolean) {
+  const className = classNames("text-sky-600 mt-4", {
+    "animate-pulse": !appLoaded,
+  });
+  const msg = appLoaded ? "JS bundle loaded!" : "... loading JS bundle ...";
```
Comment (tmeyer2115) [inline packages/studio-ui/src/AppWithLazyLoading.tsx:72]: Small nit: Can we come up with a more friendly message for non-technical users? They won't know what a JS bundle is.

File: e2e-tests/__screenshots__/darwin/add-element.spec.ts/can-add-a-Footer-component-1.png
Comment (oshi97) [reply e2e-tests/__screenshots__/darwin/add-element.spec.ts/can-add-a-Footer-component-1.png:1]: I don't think it's worth bothering with if playwright can't even detect that the screenshot has changed

File: packages/studio-ui/src/AppWithLazyLoading.tsx (hunk 5 above)
Comment (oshi97) [reply packages/studio-ui/src/AppWithLazyLoading.tsx]: updated!

**Review by alextaing — APPROVED** [review]
⚡
